index.html

```diff
@@ -6,11 +6,92 @@
     <meta name="viewport" content="width=device-width, initial-scale=1.0">
     <meta http-equiv="X-UA-Compatible" content="ie=edge">
     <title>Responsive Cake webpage</title>
-    <!-- Add a link to your css file here -->
+    <link rel="stylesheet" href="/style.css">
 </head>
 
 <body>
-    <!-- Add your markup here -->
+    <header>
+        <div class="header">
+            <div class="menu_icon">Menu
+            </div>
+            <div class="header_title">
+                <div>
+                    <h2 class="header_text">The best cakes in town delivered to your door</h2>
+                </div>
+                <div class="header_img">
+                    <a href="">
+                        <img 
+                        src="/images/cake-6364361_960_720.png"
+                        alt="Image of cupcake"
+                        class="header_img2"
+                        width="100%"
+                        />
+                    </a>
+                </div>
+            </div>
+        </div>
+    </header>
+
+    <main>
+        <article class="intro_section">
+            <div class="intro_text">
+                <h1>Welcome</h1>
+                <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum</p>
+            </div>
+            <div class="intro_img_container">
+                <a href="">
+                    <img 
+                    src="/images/birthday cake 0.jpg"
+                    alt="birthday cake with pink writing"
+                    class="intro_img"
+                    />
+                </a>
+            </div>
+        </article>
+
+        <div class="gallery_container">
+            <div class="gallery_img_container">
+                <a href="" class="gallery_img_link">
+                    <img 
+                    src="/images/front cake 0.jpg"
+                    alt="wedding cake in garden"
+                    class="gallery_img"
```

Comment:
not sure about underscore class naming , try to read about BEM naming here https://getbem.com/naming/